junit-platform-launcher/src/main/java/org/junit/platform/launcher/TestPlan.java

```diff
@@ -196,12 +196,26 @@ public Set<TestIdentifier> getChildren(String parentId) {
 	 * @throws PreconditionViolationException if no {@code TestIdentifier}
 	 * with the supplied unique ID is present in this test plan
 	 */
+	@API(status = DEPRECATED, since = "1.10")
+	@Deprecated
 	public TestIdentifier getTestIdentifier(String uniqueId) throws PreconditionViolationException {
 		Preconditions.notBlank(uniqueId, "unique ID must not be null or blank");
-		UniqueId uniqueIdObject = UniqueId.parse(uniqueId);
-		Preconditions.condition(allIdentifiers.containsKey(uniqueIdObject),
+		return getTestIdentifier(UniqueId.parse(uniqueId));
+	}
+
+	/**
+	 * Get the {@link TestIdentifier} with the supplied unique ID.
+	 *
+	 * @param uniqueId the unique ID to look up the identifier for; never
+	 * {@code null} or blank
+	 * @return the identifier with the supplied unique ID; never {@code null}
+	 * @throws PreconditionViolationException if no {@code TestIdentifier}
+	 * with the supplied unique ID is present in this test plan
+	 */
+	@API(status = MAINTAINED, since = "1.10")
+	public TestIdentifier getTestIdentifier(UniqueId uniqueId) {
```

Comment:
❌ The other methods in this class that currently accept a `String uniqueId` should also be given overloads with `UniqueId` as a parameter.